shell/src/main/java/alluxio/cli/fsadmin/metrics/ClearCommand.java

```diff
@@ -0,0 +1,259 @@
+      return -1;
+    }
+    // By default, metrics clear clears the metrics of the whole cluster
+    // If --master is passed in, clear metrics of the leading master
+    // If --workers a,b,c is passed in, clear metrics of specific workers
+    if (cl.hasOption(WORKERS_OPTION_NAME)) {
+      String workersValue = cl.getOptionValue(WORKERS_OPTION_NAME);
+      Set<String> workerSet = new HashSet<>(Arrays.asList(workersValue.split(",")));
+
+      try (FileSystemContext context = FileSystemContext.create(mAlluxioConf)) {
+        AlluxioBlockStore store = AlluxioBlockStore.create(FileSystemContext.create(mAlluxioConf));
+        List<WorkerNetAddress> addressList = store.getEligibleWorkers().stream()
+            .map(BlockWorkerInfo::getNetAddress).collect(Collectors.toList());
+        List<WorkerNetAddress> workersToClear = new ArrayList<>();
+        for (WorkerNetAddress worker : addressList) {
+          if (workerSet.contains(worker.getHost())) {
+            workersToClear.add(worker);
+            workerSet.remove(worker.getHost());
+          }
+        }
+        if (workerSet.size() != 0) {
+          System.out.printf("Cannot find workers of hostnames %s%n", String.join(",", workerSet));
+          return -1;
+        }
+        return clearWorkers(workersToClear, context) ? 0 : -1;
+      }
+    }
+
+    if (options.length == 0) {
+      try (FileSystemContext context = FileSystemContext.create(mAlluxioConf)) {
+        AlluxioBlockStore store = AlluxioBlockStore.create(FileSystemContext.create(mAlluxioConf));
+        List<WorkerNetAddress> addressList = store.getEligibleWorkers().stream()
+            .map(BlockWorkerInfo::getNetAddress).collect(Collectors.toList());
+        if (!clearWorkers(addressList, context)) {
+          return -1;
+        }
+      }
+    }
+
+    if (options.length == 0 || cl.hasOption(MASTER_OPTION_NAME)) {
+      // Clear worker metrics before master metrics since worker metrics report
+      // may be flaky during metrics clearance
+      try {
+        mMetricsClient.clearMetrics();
+        System.out.printf("Successfully cleared metrics of Alluxio leading master.%n");
+      } catch (Exception e) {
+        System.out.println("Fatal error: " + e);
+        return -1;
+      }
+    }
+    return 0;
+  }
+
+  /**
+   * Clears the metrics of a list of workers.
+   *
+   * @param workers the workers to clear metrics of
+   * @param context FileSystemContext
+   * @return true if clear succeed, false otherwise
+   */
+  private boolean clearWorkers(List<WorkerNetAddress> workers,
+      FileSystemContext context) throws IOException {
+    int workerNum = workers.size();
+    if (workerNum == 0) {
+      System.out.println("No worker metrics to clear.");
+      return true;
+    } else if (workerNum == 1) {
+      clearWorkerMetrics(workers.get(0), context);
+    } else {
+      List<Future<Void>> futures = new ArrayList<>();
+      int parallelism = Math.min(workerNum, DEFAULT_PARALLELISM);
```

Comment:
can we add a `--parallelism` option for this?

Reply:
Added the parallelism option